src/compiler/get_bloom_graph.py

```diff
@@ -0,0 +1,30 @@
+from transformers import AutoTokenizer, BloomModel, BloomConfig
+import torch
+
+config = BloomConfig()
+config.torchscript = True
+
+tokenizer = AutoTokenizer.from_pretrained("bigscience/bloom-560m")
+model = BloomModel(config)
+model.eval()
+model = BloomModel.from_pretrained("bigscience/bloom-560m", torchscript=True)
+inputs = tokenizer("Hello, my dog is cute", return_tensors="pt")
+t1 = inputs['input_ids']
+t2 = inputs['attention_mask']
+
+
+module = torch.jit.trace(model, t1)
+graph = module.graph.copy()
```

Comment:
this is how we can extract the computational graph of language models. here we extract the computational graph of the `bloom-model560m` language model.

From the computational graph we can then turn it into a intermediate language and start optimizing it and lowering it cuda :)

Reply:
Would it be possible to compile the graph directly to cuda without intermediate language?

Reply:
alternatively, could we compile to a hardware-agnostic framework like openCL?

Reply:
> Would it be possible to compile the graph directly to cuda without intermediate language?

@VictorOdede we can lower from the initial ast but the papers I’ve read use an intermediate language for machine independent optimizations.

Reply:
> alternatively, could we compile to a hardware-agnostic framework like openCL?

I think we can compile to Opencl. Python has bindings for this. So it would just be a question of turning the graph into an ast and identifying the linear algebra operations in the given language model and lowering this through an intermediate Language to either opencl or cuda. Would you prefer opencl? Please share your thoughts. Thanks

Reply:
My opinion is that we should try compiling to cuda first because we want to support the devices that most people are using atm. After we have our cuda optimizations then maybe we can also do ROCm since that's the 2nd most used platform. After that then we can consider more hardware agnostic optimizations.

Reply:
@VictorOdede yep, i think what im thinking of lowering the graph to CUDA C code

Reply:
Makes sense. What does the code do so far?

Reply:
How’s it going @VictorOdede - so far it lowers a simple linear algebra language - vector addition and matrix addition to C. 

It contains a parser, an intermediate language and a c code generator ñ.